packages/js/product-editor/src/hooks/use-number-input-props.ts

```diff
@@ -34,6 +38,11 @@ export const useNumberInputProps = ( {
 				onFocus( event );
 			}
 		},
+		onBlur( event ) {
+			if ( onBlur ) {
+				onBlur( event );
+			}
+		},
 		onKeyUp( event: React.KeyboardEvent< HTMLInputElement > ) {
 			const amount = Number.parseFloat( value || '0' );
 			const step = Number( event.currentTarget.step || '1' );
```

Comment:
We should move the stepper handling to the `onKeyDown` event. This will allow the user to quick step through values by holding down the up/down arrow.

```
		onKeyUp( event: React.KeyboardEvent< HTMLInputElement > ) {
			if ( event.code === 'ArrowUp' || event.code === 'ArrowDown' ) {
				event.preventDefault();
			}
		},
		onKeyDown( event: React.KeyboardEvent< HTMLInputElement > ) {
			if ( event.code === 'ArrowUp' || event.code === 'ArrowDown' ) {
				event.preventDefault();

				const amount = Number.parseFloat( value || '0' );
				const step = Number( event.currentTarget.step || '1' );
				if ( event.code === 'ArrowUp' ) {
					onChange( String( amount + step ) );
				}
				if ( event.code === 'ArrowDown' ) {
					onChange( String( amount - step ) );
				}
			}

			if ( onKeyUp ) {
				onKeyUp( event );
			}
		},
```